src/power_grid_model/core/buffer_handling.py

```diff
@@ -233,7 +235,7 @@ def get_buffer_view(data: np.ndarray | Mapping[str, np.ndarray], schema: Compone
 
 def create_buffer(
     properties: BufferProperties, schema: ComponentMetaData
-) -> np.ndarray | dict[ComponentType, np.ndarray]:
+) -> np.ndarray | dict[ComponentTypeVar, np.ndarray]:
```

Comment:
for return types, being specific is fine
```suggestion
) -> np.ndarray | dict[ComponentType, np.ndarray]:
```